File: iotOnGolang/internal/model/MaintenanceTicket.go
package model

import (
    "time"
    "gorm.io/gorm"
)

//==============================================================
// MaintenanceTicket Declaration
//==============================================================
type MaintenanceTicket struct {
    gorm.Model
     TicketNumber                                    string
    OpenedAt                                                            time.Time
    ClosedAt                                                            time.Time
    DeviceId         *uint
    Device           *IoTDevice `gorm:"foreignKey:DeviceId"`
    TenantId         *uint
    Tenant           *Tenant `gorm:"foreignKey:TenantId"`
    Priority                      MaintenancePriority
    Status                      MaintenanceStatus

// parent associations as their child

}

File: iotOnDjango/models/MaintenanceTicket.py

from django.db import models
from iotOnDjango.models.MaintenancePriority import MaintenancePriority
from iotOnDjango.models.MaintenanceStatus import MaintenanceStatus

#======================================================================
# Class MaintenanceTicket Declaration
#======================================================================
class MaintenanceTicket (models.Model):

#======================================================================
# attribute declarations
#======================================================================
	ticketNumber = models.CharField(max_length=200, null=True)
	openedAt = models.CharField(max_length=64, null=True)
	closedAt = models.CharField(max_length=64, null=True)
	device = models.ForeignKey('IoTDevice', on_delete=models.CASCADE, null=True, blank=True, related_name='+')
	tenant = models.ForeignKey('Tenant', on_delete=models.CASCADE, null=True, blank=True, related_name='+')
	priority = models.CharField(max_length=64, null=True, choices=[(tag.name, tag.value) for tag in MaintenancePriority])
	status = models.CharField(max_length=64, null=True, choices=[(tag.name, tag.value) for tag in MaintenanceStatus])

#======================================================================
# function declarations
#======================================================================
	def toString(self):
		str = ""
		str = str + self.ticketNumber
		str = str + self.openedAt
		str = str + self.closedAt
		str = str + self.priority
		str = str + self.status
		return str;
    
	def __str__(self):
		return self.toString();

	def identity(self):
		return "MaintenanceTicket";
    
	def objectType(self):
		return "MaintenanceTicket";
